processing/src/main/java/org/apache/druid/query/filter/vector/VectorValueMatcher.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.filter.vector;
+
+import org.apache.druid.segment.vector.VectorSizeInspector;
+
+/**
+ * A vectorized value matcher.
+ *
+ * @see org.apache.druid.query.filter.ValueMatcher, the non-vectorized version
```

Comment:
ValueMatcher doesn't have a description and ValueMatcher list of implementation classes is quite big to understand its true role clearly, I think there should be a reference to a thorough description with a showcase code explaining (in general) the supposed way of interaction of ValueMatcher, Filter, Cursor and possibly other entities. And I think all other interfaces in that chain should also contain that reference link as most of them also have no description while having a huge list of implementations.

Reply:
I added some more javadocs to ValueMatcher and VectorValueMatcher that should hopefully point readers in the right direction. Its main role is to power filtered cursors (the ones that don't use indexes, anyway) and filtered aggregators.

Fwiw, if you want to learn more about the Druid query path, the best way IMO is to try starting at QueryableIndexStorageAdapter's `makeCursors` method and spider out from there to check out what it does and what all of its callees do. That only covers the immutable-segment part of the query path (mutable realtime segments go through IncrementalIndexStorageAdapter), but the immutable-segment code is where more of the action happens anyway, so it's a better place to start. After that it is also useful to look at the callers of `makeCursors` (generally query engines) and see what they do. Usually the first thing they do is make selectors and aggregators, then start walking through and processing rows. Even if javadocs on all these interfaces were improved, this exercise would still be useful, since it'd better show how everything ties together.